src/Codeception/Lib/Generator/GherkinSnippets.php

```diff
@@ -16,22 +16,27 @@ public function {{methodName}}({{params}})
      {
         throw new \Codeception\Exception\Incomplete("Step `{{text}}` is not defined");
      }
-     
+
 EOF;
 
     protected $snippets = [];
     protected $processed = [];
 
-    public function __construct($settings)
+    public function __construct($settings, $test = null)
     {
         $loader = new Gherkin($settings);
+        if (is_null($test)) {
```

Comment:
Apparently Symfony Console handles correctly optional param if not set. If not set then `$test` will be null when passed to `GherkinSnippets` constructor.
This is confirmed by the tests that are passed so far.